cmd/serve.go

```diff
@@ -43,15 +44,19 @@ func startServer(_ *cobra.Command, _ []string) error {
 
 	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
 
-	srv, err := server.NewServer(cfg)
+	ctx, cancelFn := context.WithCancel(context.Background())
```

Comment:
We can `defer` the `cancelFn` here to make this more foolproof :)

Reply:
yes, it makes absolutely sense. And we save one line of code :)